src/renderer/wg_engine/tvgWgGeometry.cpp

```diff
@@ -65,7 +65,7 @@ void WgPolyline::appendPoint(WgPoint pt)
 {
     if (pts.count > 0) {
         float distance = pts.last().dist(pt);
-        if (distance > 0) {
+        if (!tvg::zero(distance * 1e-1)) {
```

Comment:
can you please explain why tvg::zero is not enough?

Reply:
also you are using different factors in different places for values of different meanings. can you also explain please?

Reply:
@mgrudzinska 
In the animations sometimes we can meet shapes with polylines like this:
```
    auto shape = tvg::Shape::gen();
    shape->moveTo(-1.74846e-06, -40);
    shape->lineTo(34.641, 20);
    shape->lineTo(-34.641, 20);
    shape->lineTo(4.76995e-07, -40);
    shape->close();
```
They are manually closed, but not close enough to detect it.

Reply:
I don't think this is the way to solve this. 
try this:
```
        auto shape = tvg::Shape::gen();
        shape->moveTo(-1e-05, -40);         //try also: shape->moveTo(-1.1e-05, -40); 
        shape->lineTo(34.641, 20);
        shape->lineTo(-34.641, 20);
        shape->lineTo(0, -40);
        shape->close();
        shape->strokeFill(255, 0, 0);
        shape->strokeWidth(15);
        shape->translate(200,200);
        canvas->push(std::move(shape));
```
the result is clearly not perfect for SW. and GL (but in exaclty the same way not perfect), but it works for WG. if we change this shape to start in ```shape->moveTo(-1.1e-05, -40); ``` then the results for SW and GL don't change (still not perfect in the same way), but in WG there is a different problem. 

I think that if such cases are present in  examples we have to consider changing the epsilon value, but for all engines in the same way. 
I don't understand why WG needs a different value of epsilon especially only in one place.

Reply:
you are right (private correspondence) that non of the engines rely on the epsilon values used in tvg::zero. 

I think that you may need you own epsilon, but let me test some cases and I'll come back to this

Reply:
Good idea to add own epsilon. 
Please, check the solution

Reply:
now I like it!

and I found the problem, which was bothering me, why the result is not symmetrical - https://github.com/thorvg/thorvg/pull/2743